packages/yourloops/lib/providers/patient-list.provider.hook.tsx

```diff
@@ -0,0 +1,106 @@
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import { useState } from 'react'
+import { type PatientsFilters } from './models/patients-filters.model'
+import { type PatientListContextResult } from './models/patient-list-context-result.model'
+import { PatientListColumns } from '../../components/patient-list/enums/patient-list.enum'
+import { type GridColumnVisibilityModel, useGridApiRef } from '@mui/x-data-grid'
+import { useAuth } from '../auth'
+
+const DEFAULT_FILTERS = {
+  pendingEnabled: false,
+  manualFlagEnabled: false,
+  telemonitoredEnabled: false,
+  timeOutOfTargetEnabled: false,
+  hypoglycemiaEnabled: false,
+  dataNotTransferredEnabled: false,
+  messagesEnabled: false
+}
+
+export const usePatientListProviderHook = (): PatientListContextResult => {
+  const { user, updatePreferences } = useAuth()
+  const gridApiRef = useGridApiRef()
+
+  const getColumnPreference = (ColumnName: PatientListColumns): boolean => {
+    const userPreferredColumns = user.preferences?.patientsListSortedOptionalColumns
+    return userPreferredColumns ? userPreferredColumns.includes(ColumnName) : true
+  }
+
+  const [filters, setFilters] = useState<PatientsFilters>(DEFAULT_FILTERS)
+  const [displayedColumns, setDisplayedColumns] = useState<GridColumnVisibilityModel>({
+    [PatientListColumns.Flag]: true,
+    [PatientListColumns.System]: getColumnPreference(PatientListColumns.System),
+    [PatientListColumns.Patient]: true,
+    [PatientListColumns.TimeOutOfRange]: user.isUserHcp() ? getColumnPreference(PatientListColumns.TimeOutOfRange) : false,
+    [PatientListColumns.SevereHypoglycemia]: user.isUserHcp() ? getColumnPreference(PatientListColumns.SevereHypoglycemia) : false,
+    [PatientListColumns.DataNotTransferred]: user.isUserHcp() ? getColumnPreference(PatientListColumns.DataNotTransferred) : false,
+    [PatientListColumns.LastDataUpdate]: getColumnPreference(PatientListColumns.LastDataUpdate),
+    [PatientListColumns.Messages]: user.isUserHcp() ? getColumnPreference(PatientListColumns.Messages) : false,
+    [PatientListColumns.Actions]: true
+  })
+
+  const buildColumnsPreferencesArray = (columnsVisibilityModel: GridColumnVisibilityModel): string[] => {
+    const columnsPreferences = []
+    for (const columnName in columnsVisibilityModel) {
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches to further communicate your intent.  -->
**🎯 suggestion (non-blocking):** ‏Could have been done with a reducer!